### PR TITLE
[workspace] update TypeScript to latest release

### PR DESCRIPTION
**Pull request by Simek**
# Why

It's always nice to stay up-to-date with dependencies. Currently used version is on a bit oldish side, let's update it!

# How

This PR updates the TypeScript used in the workspace and fixes reported issues.

# Test Plan

Successful run of `yarn` && `yarn build` in the workspace root.

### CODE REVIEW

File: packages/dev-tools/tsconfig.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "noImplicitAny": false,
     "outDir": "build/server",
     "rootDir": "server",
-    "allowJs": true,
+    "allowJs": false,
```
Comment (EvanBacon) [inline packages/dev-tools/tsconfig.json:16]: Would this prevent this from getting copied over? https://github.com/expo/expo-cli/blob/master/packages/dev-tools/server/babel.config.js not sure if it's needed or not though.

Comment (Simek) [reply packages/dev-tools/tsconfig.json:16]: Hmm, not sure either. I have flipped this flag to get rid of error below:
```
@expo/dev-tools: $ ts-node ./server/extract-fragment-types
@expo/dev-tools: /Users/simek/GitHub/expo-cli/node_modules/ts-node/src/index.ts:755
@expo/dev-tools:           throw new TypeError(`${relative(cwd, fileName)}: Emit skipped`)
@expo/dev-tools:                 ^
@expo/dev-tools: TypeError: ../config/build/index.js: Emit skipped
@expo/dev-tools:     at getOutput (/Users/simek/GitHub/expo-cli/node_modules/ts-node/src/index.ts:755:17)
@expo/dev-tools:     at Object.compile (/Users/simek/GitHub/expo-cli/node_modules/ts-node/src/index.ts:968:32)
```
The problem has been described earlier in `ts-node` repo, and occurs probably due to some config path issues. I will investigate this problem further, there should be better way to fix this than flipping the flag.

**Comment by Simek** [comment]
@EvanBacon What do you think about adding [`-T` (transpile-only)](https://github.com/TypeStrong/ts-node#cli-and-programmatic-options) flag to `ts-node` commands? Is it fine in context wi which those commands are used? This addition allow me to revert the questionable `tsconfig.json` change.

**Comment by brentvatne** [comment]
@Simek - let's try it!

**Pull request merged**